### PR TITLE
RFC: Switch from yarn to pnpm

### PR DESCRIPTION
**Pull request by bhollis**
This is what it would look like to switch to pnpm from yarn. I'd like to move away from Yarn v1 because it isn't maintained and it offers little over npm. Just using npm would be fine, but I wanted to try out pnpm, and with some brief investigation it looks quite nice.

* Much faster than Yarn or NPM.
* By default only exposes packages at the root of `node_modules` that you directly depend on, so you don't accidentally end up depending on one of your transitive dependencies.
* Includes some of Yarn's ergonomics (such as not requiring `npm run fix` - you can do `pnpm fix`).
* Supports all of Yarn's commands and flags (e.g. `--frozen-lockfile`, which is nicer than `npm ci` which as a side effect deletes your node_modules).
* Lots of nice options for configuring dependencies and overriding them when they're broken.
* Maybe better Windows support?

To get started, I installed `corepack` (if you're on Windows you should already have it), then ran `corepack enable`. CorePack will now automatically install the correct version of pnpm when you run a pnpm command. It'll also tell you if you forget and type `yarn` or `npm`. I also deleted node_modules so I didn't have all the old transitive dependency folders.

Installs are so fast that I'm experimenting with running an install before the devserver starts - this means you can switch branches while the devserver is running and everything comes up correctly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -248,5 +252,24 @@
...
+      "@fortawesome/fontawesome-svg-core": "*"
+    },
+    "updateConfig": {
+      "ignoreDependencies": [
+        "fork-ts-checker-notifier-webpack-plugin",
+        "fork-ts-checker-webpack-plugin",
+        "@fortawesome/fontawesome-free",
+        "@fortawesome/fontawesome-svg-core"
+      ]
+    }
```
Comment (bhollis) [inline package.json:273]: This stuff is pretty neat - you can remove things from the output of `npm outdated`, silence deprecation warnings, and correct outdated peer dependencies, among others.

**Comment by bhollis** [comment]
Looks like Jest is built on implicit dependencies that I'll have to find.

**Comment by robojumper** [comment]
I like it. The hard links and junctions work well on my (NTFS) Windows.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -56,12 +60,12 @@ More detailed information on these steps is [here](https://docs.github.com/en/ge
...
-    * If you're using PowerShell on Windows, you may need to run `Set-ExecutionPolicy -Scope CurrentUser -ExecutionPolicy Unrestricted` to allow yarn to run.
-  * Windows-based developers will need to install `windows-build-tools` (run `yarn global add windows-build-tools` in your terminal) globally prior to running `yarn install`. Refer to issue #1439 for [details](https://github.com/DestinyItemManager/DIM/issues/1439).
-  * Linux-based developers will need to install `build-essential` (`sudo apt-get install -y build-essential`) prior to running `yarn install`.
-* Run `yarn start`
+* Run `pnpm install`
+  * If `pnpm` isn't installed, run `corepack enable` and try again, or [install it manually](https://pnpm.io/installation).
+    * If you're using PowerShell on Windows, you may need to run `Set-ExecutionPolicy -Scope CurrentUser -ExecutionPolicy Unrestricted` to allow pnpm to run.
+  * Windows-based developers will need to install `windows-build-tools` (run `pnpm install -g windows-build-tools` in your terminal) globally prior to running `pnpm install`. Refer to issue #1439 for [details](https://github.com/DestinyItemManager/DIM/issues/1439).
```
Comment (delphiactual) [inline docs/CONTRIBUTING.md:66]: likely no longer needed 
image
https://github.com/felixrieseberg/windows-build-tools

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -56,12 +60,12 @@ More detailed information on these steps is [here](https://docs.github.com/en/ge
...
 Once you have cloned the repository or a fork of the repository to your local machine, in the root directory:
 
-* Run `yarn install`
-  * If `yarn` isn't installed, install it with `npm i -g yarn`.
-    * If you're using PowerShell on Windows, you may need to run `Set-ExecutionPolicy -Scope CurrentUser -ExecutionPolicy Unrestricted` to allow yarn to run.
-  * Windows-based developers will need to install `windows-build-tools` (run `yarn global add windows-build-tools` in your terminal) globally prior to running `yarn install`. Refer to issue #1439 for [details](https://github.com/DestinyItemManager/DIM/issues/1439).
-  * Linux-based developers will need to install `build-essential` (`sudo apt-get install -y build-essential`) prior to running `yarn install`.
-* Run `yarn start`
+* Run `pnpm install`
+  * If `pnpm` isn't installed, run `corepack enable` and try again, or [install it manually](https://pnpm.io/installation).
```
Comment (nev-r) [inline docs/CONTRIBUTING.md:64]:
```suggestion
  * If `pnpm` isn't installed, run `corepack enable` and try again, or [install it manually](https://pnpm.io/installation).
  * If `corepack enable` fails, try running it with Administrator privileges.
```

**Comment by nev-r** [comment]
no worse than yarn :shrug:

**Review by ryan-rushton — APPROVED** [review]
I'm on board, I have never actually tried pnpm but it gets a bit of community love and new yarn never really clicked for me.

I've experimented with yarn 2/3 a few times over the years and it have never quite gelled with it. It never seemed to bring the performance upgrade I expected and I never found the zero install feature useful for what I do.

**Pull request merged**